### PR TITLE
Add aarch64 target for darwin desktops

### LINKED ISSUE #77: Support for Apple M1 aarch64
Application-services is currently running into an issue with compiling NSS via rust-android-gradle https://github.com/mozilla/application-services/issues/4665.

I believe it's potentially something to do with `rust-android-gradle` associating Darwin = x86_64 https://github.com/mozilla/application-services/issues/4665#issuecomment-984161587 which works if the Mac is using a JDK = x86_64 but M1's are now defaulting to downloading arm64 (homebrew) JDKs which causes the builds to break.

We probably need to allow the ability for Darwin to = arm64

### PR DESCRIPTION
**Pull request by skhamis**
fixes #77 

Turned out to be a very small patch! Tested with both x86_64 and arm64 JDKs and both seem to compile nicely (the a-s issue mentioned in https://github.com/mozilla/rust-android-gradle/issues/77) . The only issue is I don't necessarily know the best way to handle the deprecation of darwin to officially split darwin to use the {-x86_64, -aarch64}. Throwing this up for early visibility....

### CODE REVIEW

**Comment by skhamis** [comment]
@ncalexan What do you think of going the darwin + darwin-{x86_64, aarch64} vs just removing the darwin target (breaking change) and only have the darwin with the modifiers? Curious your thoughts!

File: plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt [hunk 1]
```diff
@@ -25,12 +25,25 @@ val toolchains = listOf(
                 "<compilerTriple>",
                 "<binutilsTriple>",
                 "desktop/linux-x86-64"),
+        // TODO: Best way to handle deprecation here?
```
Comment (ncalexan) [inline plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt]: Fold the third commit into this one, please.

File: plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt [hunk 2]
```diff
@@ -25,12 +25,25 @@ val toolchains = listOf(
...
                 "x86_64-apple-darwin",
                 "<compilerTriple>",
                 "<binutilsTriple>",
                 "desktop/darwin"),
+        Toolchain("darwin-x86-64",
+                ToolchainType.DESKTOP,
+                "x86_64-apple-darwin",
+                "<compilerTriple>",
+                "<binutilsTriple>",
+                "desktop/darwin"),
```
Comment (ncalexan) [inline plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt:42]: These need to go elsewhere, surely.  Just `desktop/darwin-x86-64`.

File: plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt [hunk 3]
```diff
@@ -25,12 +25,25 @@ val toolchains = listOf(
...
+        Toolchain("darwin-aarch64",
+                ToolchainType.DESKTOP,
+                "aarch64-apple-darwin",
+                "<compilerTriple>",
+                "<binutilsTriple>",
+                "desktop/darwin"),
```
Comment (ncalexan) [inline plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt:48]: Ditto.

File: README.md [hunk 4]
```diff
@@ -160,8 +161,17 @@ cargo {
 
 A list of Android targets to build with Cargo; required.
 
-Valid targets are `arm`, `arm64`, `x86`, `x86_64` (Android), and `'linux-x86-64'`, `'darwin'`,
-`'win32-x86-64-gnu'`, and `'win32-x86-64-msvc'` (Desktop).
+Valid targets for **Android** are:
+
+>`'arm'`, `'arm64'`, `'x86'`, `'x86_64'`
```
Comment (ncalexan) [inline README.md:166]: This is an improvement, but let's make it way easier to copy-paste: maybe just a block with each entry on a line, or a list?

File: README.md [hunk 5]
```diff
@@ -160,8 +161,17 @@ cargo {
...
+
+Valid targets for **Desktop** are:
+> `'linux-x86-64'`,
+ `'darwin-x86-64' (Intel)`,
```
Comment (ncalexan) [inline README.md:170]: Drop the thing in parentheses that won't be copy-pasted.

File: README.md [hunk 6]
```diff
@@ -481,6 +491,19 @@ $ ls -al samples/maven-repo/org/mozilla/rust-android-gradle/org.mozilla.rust-and
 samples/maven-repo/org/mozilla/rust-android-gradle/org.mozilla.rust-android-gradle.gradle.plugin/0.4.0/org.mozilla.rust-android-gradle.gradle.plugin-0.4.0.pom
 ```
 
+# Testing Local changes
+After running `./gradlew publish` -- An easy way to locally test changes made in this plugin is to simply add this to your project's `settings.gradle`:
```
Comment (ncalexan) [inline README.md:495]: You needn't run `./gradlew publish` to use this, in fact.  The plugin will be built (and updated!) as needed.

File: README.md [hunk 7]
```diff
@@ -526,7 +549,7 @@ $ ls -al samples/library/build//outputs/aar/library-debug.aar
 
 To test in a real project, use the local Maven repository in your `build.gradle`, like:
 
-```
+```gradle
 buildscript {
     repositories {
         maven {
```
Comment (ncalexan) [inline README.md:555]: Bump the `samples/maven-repo` below to `build/local-repo` while you're here, please.

Comment (ncalexan) [reply README.md:555]: And the version number.

**Review by ncalexan — APPROVED** [review]
Just nits -- everything looks fine.  I'll land this squashed if you prefer to not squash it yourself.  I would like to see it rebased onto https://github.com/mozilla/rust-android-gradle/pull/81 once those tests are green in automation, the new tests updated to include the new Darwin targets and locations, and repushed to be sure we've got this correct.  If you have time, please do it; otherwise, I'll try to get to it shortly.

Thanks, @skhamis!

**Comment by skhamis** [comment]
@ncalexan Thanks for the review! I have applied all requested nits/changes and rebased ontop of https://github.com/mozilla/rust-android-gradle/pull/81 for those CI runs. I also squashed and bumped the plugin version -- let me know if I missed anything! Should look good once the CIs pass!

**Comment by ncalexan** [comment]
I landed this manually onto `master`: c4bf58464af845fb97bbf7b15a5f4603f3af9dca.

Sorry for the unusual landing; I had some CI testing to work through.  Thanks for pushing this forward, @skhamis!

**Comment by ncalexan** [comment]
Oh -- I cut v0.9.1 with this change as well.  Please test and make sure it works for y'all!

**Pull request closed without merging**